File: src/logicServer/src/main/java/via/sep3/logicserver/repositories/interfaces/ExerciseDAO.java
package via.sep3.logicserver.repositories.interfaces;

import java.util.List;

import via.sep3.logicserver.shared.ExerciseDTO;

public interface ExerciseDAO {
    ExerciseDTO createExercise(ExerciseDTO exercise) throws Exception;

    List<ExerciseDTO> getExercises() throws Exception;

    void editExercise(ExerciseDTO dto) throws Exception;

    void deleteExercise(int id) throws Exception;
}

File: src/dataAccessServer/Repositories/Logic/ExerciseDAO.cs
using Repositories.Interfaces;
using Shared.DTOs;

namespace Repositories.Logic;

public class ExerciseDAO : IExerciseDAO
{
    public Task<ExerciseDTO> CreateExercise(ExerciseDTO exercise)
    {
        // Access to database

        return Task.FromResult(exercise);
    }

    public Task<IEnumerable<ExerciseDTO>> GetExercises()
    {
        throw new NotImplementedException();
    }
}